### DISCUSSION TITLE
Size comparison to react-password-strength-bar library

### DISCUSSION THREAD

**Discussion by mfialko** [zxcvbn-ts/zxcvbn#292] [Q&A] [answered]
Good day!

Recently, I have been using the [react-password-strength-bar](https://www.npmjs.com/package/react-password-strength-bar) library to calculate password strength, which uses the original **zxcvbn** under the hood. I switched to **zxcvbn-ts**, as it is supposed to be a modern and optimised alternative, but it adds significantly more to the bundle size than the original, outdated library. Why is that? Am I doing something wrong?

Bundle analyser. 
**zxcvbn**:
<img width="1060" alt="Screenshot 2024-12-02 at 11 09 29" src="https://github.com/user-attachments/assets/d30b1a4a-f793-4a0d-a936-be856f45c0c3">

**zxcvbn-ts**:
<img width="497" alt="Screenshot 2024-12-02 at 11 12 15" src="https://github.com/user-attachments/assets/33e5267c-eaae-4d4c-a877-e1c16bc1e03c">

**Comment by MrWook** [comment, marked as answer]
Hey, good comparison!

Zxcvbn-ts doesn't have the same dictionaries as the original one but overall they are more up to date. For example the wikipedia list from the original one is from 2016 and the one from zxcvbn-ts is from 2024.  There are a ton of new words in it.

But we are investigating means to reduce the bundle size of dictionaries as you can see in https://github.com/zxcvbn-ts/zxcvbn/issues/254
